design/common/separate-git-repository-for-helm-charts.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Separate git Repository for Helm Charts
+target-version: release-1.6
+---
+
+# Separate git Repository for Helm Charts
+
+## Summary
+
+### Goals
+
+The goal of this design document / change is to use a separate git repository for Helm charts for the Rook ecosystem.
+The Rook ecosystem Helm charts include but not limited to the Rook operators and Rook custom resources.
+
+To know this goal has been achieved, a separate repository with the current Rook operator Helm charts and upcoming Helm charts should have been created.
+In addition to the repository, CI workflows should be established to be able to release the Helm charts when needed.
+
+### Non-Goals
+
+The design document is not about introducing new Helm Charts, but to move the Helm charts development to a separate repository.
+
+## Proposal details
```

Comment:
A separate repo means that the helm charts could change independently of the Rook repo. Does this mean that we could release a new helm chart every time a fix is made to the helm chart, and we would re-generate a new helm chart for the same rook version?

Reply:
Yes, exactly. We should see the Helm charts independent of Rook itself, but only loosely "coupled" by the "recommended to be used with Rook version X.Y" that we can add to Helm charts and documentation.

Reply:
At the same time, this would be challenging. If a new property is added to a CRD, an update to the helm chart must be made to support it, and we couldn't release the new helm chart until the new rook release is available with that change. We would really have to think about the helm chart versioning, instead of not worrying about it, unless I'm overthinking this...

If we keep the sources in the same repo, it keeps us safe from any versioning headaches. Do we really need that flexibility? The more important question seems to be where we're publishing and considering how we're publishing to channels.

Reply:
Are there any advantages to having the Helm charts on a different release cadence to Rook? If not, I tend to agree with Travis that keeping them coupled saves a lot of complexity.

It might mean, however, that if we need to make a minor change to a chart that the patch-version of rook is also bumped. For example, rook 1.5.0 might be released with a helm chart (also versioned 1.5.0) and when a change is required due to a regression/bug in the chart both the chart and rook will be re-released to 1.5.1. That way if you have version 1.5.1 of the chart you know the exact version of rook you're running and vice-versa.